### PR TITLE
Feat implement backup to storage node js

### PR DESCRIPTION
**Pull request by henrhie**
Could not generate an in-memory file to be sent to the storage API
The function generates the CSV file on disk and sends it to the storage API.

### CODE REVIEW

**Review by Meldiron — CHANGES_REQUESTED** [review]
Something is not right, it shows 217 files changed. I believe node_modules s included, can we please make sure to exclude this folder?

File: nodejs/backup-to-storage/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# 📧 Backup to Storage
+
+A sample NodeJS Cloud Function for creating a backup of every document in every collecion of your database in Cloud Storage.
```
Comment (christyjacob4) [inline nodejs/backup-to-storage/README.md:3]:
```suggestion
A sample NodeJS Cloud Function for creating a backup of every document in every collection of your database in Cloud Storage.
```

File: nodejs/backup-to-storage/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+## 📝 Environment Variables
+
+Go to Settngs tab of your Cloud Function. Add the following enviromnent variables.
+
+- **APPWRITE_ENDPOINT** - Your Appwrite Endpoint
+- **APPWRITE_API_KEY** - Your Appwrite API key with `collections.read`, `files.write`, `documents.read` permissions
+- **APPWRITE_FUNCTION_PROJECT_ID** - Your Appwrite Project ID
```
Comment (christyjacob4) [inline nodejs/backup-to-storage/README.md:11]:
```suggestion
```

You can remove this env var. This is passed to the runtime by default

File: nodejs/backup-to-storage/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## 🚀 Building and Packaging
+
+To package this example as a cloud function, follow these steps
+
+```bash
+$ cd demos-for-functions/nodejs/backup-to-storage
+
+$npm install
```
Comment (christyjacob4) [inline nodejs/backup-to-storage/README.md:20]:
```suggestion
$ cd demos-for-functions/nodejs/backup-to-storage
$ npm install
```

File: nodejs/backup-to-storage/README.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+```
+.
+├── index.js
+├── utils.js
+├── node_modules/
+├── package-lock.json
+└── package.json
+```
+
+- create a tarfile using the command below:
```
Comment (christyjacob4) [inline nodejs/backup-to-storage/README.md:34]:
```suggestion
- Create a tarfile using the command below:
```

File: nodejs/backup-to-storage/README.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+```bash
+$ cd ..
+
+$ tar -zcvf code.tar.gz backup-to-storage
```
Comment (christyjacob4) [inline nodejs/backup-to-storage/README.md:39]:
```suggestion
$ cd ..
$ tar -zcvf code.tar.gz backup-to-storage
```

File: nodejs/backup-to-storage/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+```
+
+- Navigate to the Overview Tab of your Cloud Function > Deploy Tab(scroll down a bit)
```
Comment (christyjacob4) [inline nodejs/backup-to-storage/README.md:42]:
```suggestion
- Navigate to the Overview Tab of your Cloud Function > Deploy Tab (scroll down a bit)
```

**Review by christyjacob4 — CHANGES_REQUESTED** [review]
Please address the comments 👍  Rest looks great ⭐

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+is%3Aissue+label%3Ahacktoberfest+created%3A%3E2021-01-01+is%3Aopen&type=issues) and help make Appwrite better :)

**Pull request merged**